File: leetcode/lc-us/cpp/1817.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> findingUsersActiveMinutes(vector<vector<int>> &logs, int k) {
        vector<int> res(k);
        unordered_map<int, unordered_set<int>> map;
        for (auto &log: logs) {
            map[log[0]].insert(log[1]);
        }
        for (auto &x: map) {
            res[x.second.size() - 1]++;
        }
        return res;
    }
};

File: leetcode/lc-us/py/1817.py
from collections import defaultdict

from typing import List


class Solution:
    def findingUsersActiveMinutes(self, logs: List[List[int]], k: int) -> List[int]:
        res = [0] * k
        m1 = defaultdict(set)
        for idx, time in logs:
            m1[idx].add(time)
        for k, v in m1.items():
            res[len(v) - 1] += 1
        return res
